File: 1071_greatest_common_divisor_of_strings.py

def gcdOfStrings(str1: str, str2: str) -> str:
    smallest_str = str1 if len(str1) < len(str2) else str2
    smallest_len = len(smallest_str)

    for i in range(smallest_len, 0, -1):
        if len(str1) % i == 0 and len(str2) % i == 0:
            divisor = smallest_str[:i]
            if divisor * int((len(str1) / i)) == str1 and divisor * int((len(str2) / i)) == str2:
                return divisor

    return ""

File: go/1071_greatest_common_divisor_of_strings.go
package main

import (
	"strings"
)

func gcdOfStrings(str1 string, str2 string) string {
	smallerStr := str1
	if len(str2) < len(str1) {
		smallerStr = str2
	}

	for i := len(smallerStr); i >= 1; i-- {
		if len(str1)%i == 0 && len(str2)%i == 0 {
			div := smallerStr[:i]
			if str1 == strings.Repeat(div, len(str1)/i) && str2 == strings.Repeat(div, len(str2)/i) {
				return div
			}
		}
	}

	return ""
}
